### PR TITLE
OCRST: Support reboot into firmware

### PR DESCRIPTION
**Pull request by zhen-zen**
Based on 
[audk/ShellPkg/Library/UefiShellLevel2CommandsLib/Reset.c](https://github.com/acidanthera/audk/blob/master/ShellPkg/Library/UefiShellLevel2CommandsLib/Reset.c)@[f675a1b](https://github.com/acidanthera/audk/commit/f675a1b42235b508e17220878453b4b1f04dd990)

### CODE REVIEW

File: Application/ResetSystem/ResetSystem.c [hunk 1]
```diff
@@ -42,6 +46,39 @@ UefiMain (
...
+    DataSize = sizeof (OsIndications);
+    Status = gRT->GetVariable (
+      EFI_OS_INDICATIONS_SUPPORT_VARIABLE_NAME, &gEfiGlobalVariableGuid,
+      &Attr, &DataSize, &OsIndications
+      );
+    if (!EFI_ERROR (Status)) {
+      if ((OsIndications & EFI_OS_INDICATIONS_BOOT_TO_FW_UI) != 0) {
+        DataSize = sizeof (OsIndications);
+        Status = gRT->GetVariable (
+          EFI_OS_INDICATIONS_VARIABLE_NAME, &gEfiGlobalVariableGuid,
```
Comment (vit9696) [inline Application/ResetSystem/ResetSystem.c:60]: Could you please reformat this and the other calls to GetVariable/SetVariable with the usual EDK II codestyle (every argument on a separate line).

File: Application/ResetSystem/ResetSystem.c [hunk 2]
```diff
@@ -42,6 +46,39 @@ UefiMain (
...
+        }
+        Status = gRT->SetVariable (
+          EFI_OS_INDICATIONS_VARIABLE_NAME, &gEfiGlobalVariableGuid,
+          EFI_VARIABLE_NON_VOLATILE | EFI_VARIABLE_BOOTSERVICE_ACCESS | EFI_VARIABLE_RUNTIME_ACCESS,
+          sizeof (OsIndications), &OsIndications
+          );
+      }
+    }
+    if (EFI_ERROR (Status)) {
+      DEBUG ((DEBUG_WARN, "OCRST: Failed to enter firmware, maybe not supported?\n"));
```
Comment (vit9696) [inline Application/ResetSystem/ResetSystem.c:76]: Maybe print the `Status` value with `%r` at the very least?

File: Application/ResetSystem/ResetSystem.c (hunk 1 above)
Comment (zhen-zen) [reply Application/ResetSystem/ResetSystem.c]: Reformatted according to EDK II C Coding Standards Specification 2.1 5.2.2.4. The change has been squashed into previous commit.

File: Application/ResetSystem/ResetSystem.c [hunk 3]
```diff
@@ -42,6 +46,47 @@ UefiMain (
...
+          OsIndications = EFI_OS_INDICATIONS_BOOT_TO_FW_UI;
+        }
+        Status = gRT->SetVariable (
+                        EFI_OS_INDICATIONS_VARIABLE_NAME,
+                        &gEfiGlobalVariableGuid,
+                        EFI_VARIABLE_NON_VOLATILE | EFI_VARIABLE_BOOTSERVICE_ACCESS | EFI_VARIABLE_RUNTIME_ACCESS,
+                        sizeof (OsIndications),
+                        &OsIndications
+                        );
+      }
```
Comment (vit9696) [inline Application/ResetSystem/ResetSystem.c:81]: Just realised that we want `} else {` with some warning and `Status = EFI_UNSUPPORTED` when `EFI_OS_INDICATIONS_BOOT_TO_FW_UI` is not set. Sorry for not noticing earlier.

**Pull request merged**